src/seedu/addressbook/data/person/ReadOnlyPerson.java

```diff
@@ -21,6 +21,8 @@
      */
     Set<Tag> getTags();
 
+    String getAsTextNoRestrictions();
```

Comment:
You may have the implementation here rather than in Person since printing does not mutate the data (Any behaviour that does not mutate the Person can be kept in ReadOnlyPerson)